packages/frontend/src/context/UI.ts

```diff
@@ -5,16 +5,29 @@ export class UI {
     loadingPromise
 
     hasBanner: boolean = true
+    showBackBtn: boolean = false
 
     constructor() {
         makeObservable(this, {
             hasBanner: observable,
+            showBackBtn: observable,
         })
         if (typeof window !== 'undefined') {
             this.loadingPromise = this.load()
         } else {
             this.loadingPromise = Promise.resolve()
         }
+
+        window.onscroll = () => {
```

Comment:
This logic is really specific to the component that is using it. Instead of having a `showBackBtn` variable here, we should have an observable `scrollTop` variable

```
window.onscroll = () => {
  this.scrollTop = document.documentElement.scrollTop
}
```

Then in the component render logic
```
{context.ui.scrollTop > 104 + window.innerHeight / 2 ? <RenderButton /> : null}
```

This way many component can use the listener without needing custom variables in the context.